doc/guides/west/build-flash-debug.rst

```diff
@@ -27,88 +27,175 @@ directly delegate to West.
 
-The ``build`` command allows you to build any source tree from any directory
-in your file system, placing the build results in a folder of your choice.
+The ``build`` command helps you build Zephyr applications from source. You can
+use :ref:`west config <west-config-cmd>` to configure its behavior.
 
-In its simplest form, the command can be run by navigating to the root folder
-(i.e. the folder containing a :file:`CMakeLists.txt` file) of the Zephyr
-application of your choice and running::
+This command attempts to "do what you mean" when run from a Zephyr application
+source or build directory:
+
+- When running ``west build`` in an existing build directory, the board, source
+  directory, etc. are obtained from the CMake cache, and that build directory
+  is re-compiled.
+
+- The same is true if a Zephyr build directory named :file:`build` exists in
+  your current working directory.
+
+- Otherwise, the source directory defaults to the current working directory, so
+  running ``west build`` from a Zephyr application's source directory compiles
+  it.
+
+Basics
+======
+
+The easiest way to use it is to go to an application's root directory (i.e. the
+folder containing its :file:`CMakeLists.txt`) and then run::
 
   west build -b <BOARD>
 
 Where ``<BOARD>`` is the name of the board you want to build for. This is
 exactly the same name you would supply to CMake if you were to invoke it with:
 ``cmake -DBOARD=<BOARD>``.
-A build folder named :file:`build` (default name) will be created and the
-build output will be placed in it.
 
-To specify the build directory, use ``--build-dir`` (or ``-d``)::
+.. tip::
 
-  west build -b <BOARD> --build-dir path/to/build/directory
+   You can use the :ref:`west boards <west-boards>` command to list all
+   supported boards.
 
-Since the build directory defaults to :file:`build`, if you do not specify
-a build directory but a folder named :file:`build` is present, that will be used,
-allowing you to incrementally build from outside the :file:`build` folder with
-no additional parameters.
+A build directory named :file:`build` will be created, and the application will
+be compiled there after ``west build`` runs CMake to create a build system in
+that directory. If you run ``west build`` with an existing build directory, the
+application is incrementally re-compiled without re-running CMake (you can
+force CMake to run again with ``--cmake``).
 
-.. note::
-  The ``-b <BOARD>`` parameter is only required when there is no CMake cache
-  present at all, usually when you are building from scratch or creating a
-  brand new build directory. After that you can rebuild (even with additional
-  parameters) without having to specify the board again. If you're unsure
-  whether ``-b`` is required, just try leaving it out. West will print an
-  error if the option is required and was not given.
+You don't need to give ``-b`` if you've already got an existing build
```

Comment:
change
```
give ``-b``
```
to
```
use the ``-b <BOARD>`` parameter

Reply:
I think "option" would be better than "parameter" in this case. Parameters generally are used when referring to functions rather than command line programs.